src/js/stores/MesosStateStore.js

```diff
@@ -164,7 +169,10 @@ class MesosStateStore extends GetSetBaseStore {
       return null;
     }
 
-    return new Task(foundTask);
+    let result = new Task(foundTask);
+    taskCache[taskID] = result;
```

Comment:
I caught up offline and understand that it has to do with future-proofing. I'll do it and leave a clarifying comment.